### PR TITLE
Add multi thread for content manager

### PR DESCRIPTION
**Pull request by Zangetsu38**
- content manager: add multi thread support for get size. (fix freeze on open app)
- gui: fix crash if data font is missing.
- reworks delete app for can using same function inside context menu/content manager.

# Result : 
 - Delete on content manager.
image

- On App selector.
image

### CODE REVIEW

File: vita3k/gui/src/content_manager.cpp [hunk 1]
```diff
@@ -83,94 +85,119 @@ struct SaveData {
...
+                save_data_list.push_back({ get_app_index(gui, title_id)->title, title_id, save_data_size, date });
             }
+        }
+        space["savedata"] = save_data_list_size ? get_unit_size(save_data_list_size) : "-";
+        finished_get_size["savedata"] = true;
+        std::sort(save_data_list.begin(), save_data_list.end(), [](const SaveData &sa, const SaveData &sb) {
+            return sa.title < sb.title;
+        });
+    });
+    get_save_data_size.detach();
```
Comment (sunho) [inline vita3k/gui/src/content_manager.cpp:125]: You cant use detach here. We need to properly clean the states for this case.

File: vita3k/gui/src/content_manager.cpp [hunk 2]
```diff
@@ -83,94 +85,119 @@ struct SaveData {
...
+static std::map<std::string, std::string> space;
+static std::map<std::string, bool> contents_selected, finished_get_size;
+static size_t apps_list_size, save_data_list_size;
 
 static void get_save_data_list(GuiState &gui, HostState &host) {
     host.io.user_id = fmt::format("{:0>2d}", host.cfg.user_id);
     save_data_list.clear();
+    save_data_list_size = 0;
+
+    std::thread get_save_data_size([&gui, &host]() {
```
Comment (sunho) [inline vita3k/gui/src/content_manager.cpp:97]: Full of race conditions. You wouldn't want to pull in the host here.

**Review by sunho — CHANGES_REQUESTED** [review]

File: vita3k/gui/src/content_manager.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: I have no understund code works fine here 🤔

File: vita3k/gui/src/content_manager.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:97]: No understund, i need host here

File: vita3k/gui/src/content_manager.cpp (hunk 1 above)
Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: It can break randomly.

Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: Fir niw after litteraly 100 lunch never get any issue here no really understund what you want exatcly 🤔

Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: Study race condition. Otherwise, you wont understand.

Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: Oh btw for detach, its related to exitting in the middle. The thread can be spawned multiple tunes.

Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: Which can mess up the things.

Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: No sur understund here, it run function one time and after exit it so it no can run again this detach 🤔

File: vita3k/gui/src/content_manager.cpp (hunk 2 above)
Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:97]: I mean some functions are not thread safe. You should consider it.

File: vita3k/gui/src/content_manager.cpp (hunk 1 above)
Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: If it runs once, then the size is not calculated again unless restarting the emu?

Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: Run again so close content manager, etbre open it re calcule it yes, is for that value is cleaned before

Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:125]: Thread will not exit even if you close content manager.

File: vita3k/gui/src/content_manager.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:97]: Yes i have see that forbsome thinks

File: vita3k/gui/src/content_manager.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: Yes i know hmm i need thinks, you know how kill thread if run it is not finissed in moment  close content manager?

Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:125]: i have founded and fix that, now it stop thread in moment exit content manager

Comment (sunho) [reply vita3k/gui/src/content_manager.cpp:131]: That's also not thread safe and hacky. I'll just send a commit that fixes the issues.

Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:131]: oki, i have re force push

File: vita3k/gui/src/content_manager.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/gui/src/content_manager.cpp:109]: i have added this mutex if you fcan check again please

**Comment by ArbestRi** [comment]
Please close this pull request, this didn't get a update since March 2021

**Comment by Zangetsu38** [comment]
> Please close this pull request, this didn't get a update since March 2021

No yet, i want finish it soon

**Comment by ArbestRi** [comment]
Still:
Please close this PR, this didn't get a update since March 2021

**Comment by Zangetsu38** [comment]
Nop 🗿

**Comment by Zangetsu38** [comment]
Need just little reworks code 🗿

**Comment by CoffeeBrewer64** [comment]
> Still: Please close this PR, this didn't get a update since March 2021

Why? This is fine as is. As Zang said, it needs some fixing but otherwise it is fine.
